### ISSUE TITLE
Ensure that state.matches allows for object-based checks on states with no children

### ISSUE THREAD

**Issue by mattpocock** [statelyai/xstate-tools#136]
I.e. in this machine:

```ts
const machine = createMachine({
  initial: 'a',
  states: { a: {} },
})
```

You should be able to run:

`state.matches({ a: {} })`

**Comment by Andarist** [comment]
Is this desired though? It somewhat looks incomplete and/or implies that some child states can be added to it.

**Comment by mattpocock** [comment]
Yes, because you want to be able to make parallel-based checks, like so:

https://discord.com/channels/795785288994652170/962341003266760786/962343455135264828

![image](https://user-images.githubusercontent.com/28293365/162699353-50e54886-9f38-4581-a170-8c3f9eb285db.png)

**Comment by Andarist** [comment]
Yuck 😬 I don't have a better idea though - so ye, we should support this then

**Comment by davidkpiano** [comment]
Agree; this should be supported
